### PR TITLE
Vector3

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: course/include/isometry.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  double & x() {return x_;};
+  double & y() {return y_;};
+  double & z() {return z_;};
+
+  const double & operator [] (const int index) const;
+  double & operator [] (const int index);
+  Vector3 operator += (const Vector3 & obj) const;
+  Vector3 operator -= (const Vector3 & obj) const;
+  Vector3 operator *= (const Vector3 & obj) const;
+  Vector3 operator /= (const Vector3 & obj) const;
```
Comment (glpuga) [inline course/include/isometry.h]: None of these operators with assignment can be `const`. Why?

File: course/include/isometry.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  Vector3 operator + (const Vector3 & obj) const {return *this += obj;};
+  Vector3 operator - (const Vector3 & obj) const {return *this -= obj;};
+  Vector3 operator * (const Vector3 & obj) const {return *this *= obj;};
+  Vector3 operator / (const Vector3 & obj) const {return *this /= obj;};
+  bool operator == (const std::initializer_list<double> & obj) const;
+  bool operator == (const Vector3 & obj) const;
+  bool operator != (const std::initializer_list<double> & obj) const {return !(*this == obj);};
+  bool operator != (const Vector3 & obj) const {return !(*this == obj);};
+  Vector3 operator *= (const double & obj) const;
+  Vector3 operator /= (const double & obj) const;
```
Comment (glpuga) [inline course/include/isometry.h]: Same, this cannot be `const`.

File: course/src/isometry.cc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  //Vector3
+
+  const Vector3 Vector3::kUnitX(1., 0., 0.);
+  const Vector3 Vector3::kUnitY(0., 1., 0.);
+  const Vector3 Vector3::kUnitZ(0., 0., 1.);
+  const Vector3 Vector3::kZero(0., 0., 0.);
+
+  Vector3 Vector3::operator += (const Vector3 & obj) const
+  {
+    return Vector3( x_ + obj.x(), y_ + obj.y(),z_ +  obj.z());
```
Comment (glpuga) [inline course/src/isometry.cc]: This does not change do the assignation step of the operation.

File: course/src/isometry.cc [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  }
+
+  Vector3 Vector3::operator -= (const Vector3 & obj) const
```
Comment (glpuga) [inline course/src/isometry.cc]: Same here and others.

**Review by glpuga — CHANGES_REQUESTED** [review]
Some questions to answer.

File: course/include/isometry.h (hunk 1 above)
Comment (Lobotuerk) [reply course/include/isometry.h]: A const object cant use =/ or any of those, as a const cant be modified, and =/ modifiy the object

**Review by glpuga — APPROVED** [review]
Awesome! LGTM! Please merge an rebase your remaining PRs on top of master.

**Pull request merged**